qiime/plugin/testing.py

```diff
@@ -84,3 +90,25 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transform_format(self, source_format, target, filenames):
+        # Guard any non-QIIME2 Format sources from being tested
+        self.assertTrue(issubclass(source_format, FormatBase))
+
+        transformer = self.get_transformer(source_format, target)
+
+        for filename in filenames:
+            filepath = self.get_data_path(filename)
+            shutil.copy(filepath, self.temp_dir.name)
+        input = source_format(self.temp_dir.name, mode='r')
```

Comment:
`self.temp_dir.name` works for `DirectoryFormat`, but we need to provide the path to the actual file for `FileFormat`.

@ebolyen: I had a memory that creating a `FileFormat` with the path to the file's parent directory worked, but I must have been mistaken. We have this behavior for `SingleFileDirectoryFormat` (specify either the surrounding dir or the single file itself) --- does it make sense to support this type of construction for `FileFormat`, too?